src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Arm.java

```diff
@@ -15,148 +15,98 @@
-        GoalPos.STORED[1].wristPos = SmartDashboard.getNumber("StoredWristCone", GoalPos.STORED[1].wristPos);
-        GoalPos.SHELF[1].armPos = SmartDashboard.getNumber("ShelfArmCone", GoalPos.SHELF[1].armPos);
-        GoalPos.SHELF[1].wristPos = SmartDashboard.getNumber("ShelfWristCone", GoalPos.SHELF[1].wristPos);
-        GoalPos.SUBSTATION[1].armPos = SmartDashboard.getNumber("SubstationArmCone", GoalPos.SUBSTATION[1].armPos);
-        GoalPos.SUBSTATION[1].wristPos = SmartDashboard.getNumber("SubstationWristCone", GoalPos.SUBSTATION[1].wristPos);
-        GoalPos.INTAKE[1].armPos = SmartDashboard.getNumber("IntakeArmCone", GoalPos.INTAKE[1].armPos);
-        GoalPos.INTAKE[1].wristPos = SmartDashboard.getNumber("IntakeWristCone", GoalPos.INTAKE[1].wristPos);
-
-        MAX_FF_VEL = SmartDashboard.getNumber("Max FF Vel", MAX_FF_VEL);
-        MAX_FF_ACCEL = SmartDashboard.getNumber("Max FF Accel", MAX_FF_ACCEL);
-    }
-
-    public void setArmVoltage(double volts) {
+        getArmControlOnSmartDashboard();
+    }
+ 
+    public void driveArm(double vel, double accel, boolean usePID) {
+        double armFeedVolts = getKg() * getCoM().getAngle().getCos() + armFeed.calculate(vel, accel);
+        double armPIDVolts = armPID.calculate(armEncoder.getPosition(), goalPosRad[ARM]);
+        if (getArmPos() > ARM_UPPER_LIMIT_RAD || getArmPos() < ARM_LOWER_LIMIT_RAD) {
+            armFeedVolts = getKg() * getCoM().getAngle().getCos() + armFeed.calculate(0, 0);
+        }
+        // TODO: REMOVE WHEN DONE WITH TESTING (ANY CODE REVIEWERS, PLEASE REJECT MERGES
+        // TO MASTER IF THIS IS STILL HERE)
+        SmartDashboard.putNumber("ArmFeedVolts", armFeedVolts);
+        SmartDashboard.putNumber("ArmPIDVolts", armPIDVolts);
+        double volts = usePID ? armFeedVolts + armPIDVolts : armFeedVolts;
+        SmartDashboard.putNumber("ArmTotalVolts", volts);
         armMotor.setVoltage(volts);
     }
-    public void setWristVoltage(double volts) {
+
+    public void driveWrist(double vel, double accel, boolean usePID) {
+        double wristFeedVolts = wristFeed.calculate(getWristPosRelativeToGround(), vel, accel);
+        double wristPIDVolts = wristPID.calculate(getWristPos(), goalPosRad[WRIST]);
+        if (getWristPos() > WRIST_UPPER_LIMIT_RAD || getWristPos() < WRIST_LOWER_LIMIT_RAD) {
+            wristFeedVolts = wristFeed.calculate(getWristPosRelativeToGround(), 0, 0);
+        }
+        // TODO: REMOVE WHEN DONE WITH TESTING (ANY CODE REVIEWERS, PLEASE REJECT MERGES
+        // TO MASTER IF THIS IS STILL HERE)
+        SmartDashboard.putNumber("WristFeedVolts", wristFeedVolts);
+        SmartDashboard.putNumber("WristPIDVolts", wristPIDVolts);
+        double volts = usePID ? wristFeedVolts + wristPIDVolts : wristFeedVolts;
+        SmartDashboard.putNumber("WristTotalVolts", volts);
         wristMotor.setVoltage(volts);
     }
```

Comment:
See my comments on `driveArm`

Reply:
"This means if we ever go out of range that the arm will be held in position. Can you add a velocity check so that it moves if we try to move it back into range."

Reply:
[Fixed]